cluster.go

```diff
@@ -192,28 +199,40 @@ func (n *clusterNode) Close() error {
 	return n.Client.Close()
 }
 
+type CNode interface {
+	Client() *Client
+}
+
+// latency funcs return true if alls well, false if they should be marked as failing
+func defaultLatencyFunc(c *Client) bool {
+	cmd := c.Ping(context.Background())
+	if isLoadingError(cmd.err) {
+		return false
+	}
+	return true
+}
+
 func (n *clusterNode) updateLatency() {
 	const numProbe = 10
 	var dur uint64
 
-	var i int
-	for i = 0; i < numProbe; i++ {
+	for i := 0; i < numProbe; i++ {
 		time.Sleep(time.Duration(10+rand.Intn(10)) * time.Millisecond)
 
 		start := time.Now()
-		cmd := n.Client.Ping(context.TODO())
-		// sometimes failures can be really fast and actually make them the fastest node
-		if cmd.err != nil {
-			if isLoadingError(cmd.err) || isConnectionError(cmd.err) {
-				internal.Logger.Printf(context.Background(), "latency marking as failing %s %s", n.String(), cmd.err)
-				n.MarkAsFailing()
-				break
-			}
+		latencyHealthFunc := defaultLatencyFunc
+		if n.Client.Options().LatencyHealthFunc != nil {
```

Comment:
can you check if LatencyHealthFunc if nil then set to defaultLatencyFunc at init time itself?

Reply:
a good optimization